feature-libs/customer-ticketing/components/customer-ticketing-messages/customer-ticketing-messages.component.ts

```diff
@@ -1,112 +1,110 @@
-      id: 'CLOSE',
-      name: 'Close',
-    },
-    subject: 'test ticket again',
-    ticketCategory: {
-      id: 'COMPLAINT',
-      name: 'Complaint',
-    },
-    ticketEvents: [
-      {
-        author: 'Mark Rivers',
-        createdAt: '2022-06-22T20:25:02+0000',
-        message: 'This is the way',
-        attachments: [
-          {
-            filename: 'screenshot.png',
-            URL: 'https://ccv2.domain.com/occ/v2/electronics/users/0001/tickets/0013/events/0007PC/attachments/0034-034-24589',
-          },
-          {
-            filename: 'screenshot.png',
-            URL: 'https://ccv2.domain.com/occ/v2/electronics/users/0001/tickets/0013/events/0007PC/attachments/0034-034-24589',
-          },
-        ],
-      },
-      {
-        author: 'Mark Rivers',
-        createdAt: '2022-06-22T14:37:15+0000',
-        message: 'A message to consider',
-      },
-      {
-        addedByAgent: true,
-        createdAt: '2022-06-22T20:25:02+0000',
-        message: 'This is the way',
-      },
-      {
-        addedByAgent: true,
-        createdAt: '2022-06-22T20:25:02+0000',
-        message: 'This is the way',
-      },
-      {
-        author: 'Mark Rivers',
-        createdAt: '2022-06-22T14:37:15+0000',
-        message: 'A message to consider',
-      },
-    ],
-  });
+export class CustomerTicketingMessagesComponent implements OnDestroy {
+  ticketDetails$: Observable<TicketDetails | undefined> =
+    this.customerTicketingFacade.getTicket();
 
-  constructor(protected customerTicketingConfig: CustomerTicketingConfig) {}
+  constructor(
+    protected customerTicketingConfig: CustomerTicketingConfig,
+    protected customerTicketingFacade: CustomerTicketingFacade,
+    protected eventService: EventService
+  ) {}
+
+  subscription = new Subscription();
 
   messageEvents$: Observable<Array<MessageEvent> | undefined> =
     this.prepareMessageEvents();
 
   messagingConfigs: MessagingConfigs = this.prepareMessagingConfigs();
 
-  onSend(_event: { files: FileList | undefined; message: string }) {
-    // call to submit new event and upload attachment
+  onSend(event: { files: FileList | undefined; message: string }) {
+    this.subscription.add(
+      this.customerTicketingFacade
+        .createTicketEvent(this.prepareTicketEvent(event.message))
+        .subscribe((createdEvent: TicketEvent) => {
+          if (event.files?.length && createdEvent.code) {
+            this.customerTicketingFacade.uploadAttachment(
+              event.files.item(0),
+              createdEvent.code
+            );
+          } else {
+            this.eventService.dispatch({ status }, TicketEventCreatedEvent);
+          }
+        })
+    );
+  }
+
+  downloadAttachment(event: {
+    messageCode: string;
+    attachmentId: string;
+    fileName: string;
+  }) {
+    this.subscription.add(
+      this.customerTicketingFacade
+        .downloadAttachment(event.messageCode, event.attachmentId)
+        .subscribe((data) => {
```

Comment:
I'm confused here, please clarify.
Isn't the DownloadAttachment returning a Blob? how that is transformed into the downloadURL?
The download URL must be built from the input params:
downloadURL = ...adapter. getDownloadAttachmentEndpoint(customerId, ticketId, EventCode, AttachmentId)

Reply:
We need to do this In order to download the blob in a desirable format, I tried this way and it opens the raw dat in unreadable format.